src/dataframe/dataframe.jl

```diff
@@ -394,65 +393,78 @@ end
 """
     setindex!(df::DataFrame, v, col_ind::Union{Integer, Symbol})
 
-If `v` is not an `AbstractVector` then column `col_ind` is filled with `v`.
-Otherwise `v` is used as a column vector (no data copying is performed).
-`col_ind` can be either a `Symbol` or `Integer` (except `Bool`).
-If `col_ind` is present in a `DataFrame` then:
-* if `v` is an abstract vector then it replaces `df[col_ind]`
-* if `v` i not an abstract vector then existing `df[col_ind]` is filled with `v`
+Assign `v` to column `col_ind` of data frame `df`.
+
+If column `col_ind` is already present in `df` then:
+* if `v` is an `AbstractVector` then it replaces `df[col_ind]` (no copy is made)
+* otherwise, the existing `df[col_ind]` vector is filled with `v`
```

Comment:
Is this behavior really desirable? `df[col_ind]` accesses the column vector, so it sounds like assigning to this should replace the vector, not fill it with values.

Often the best rules are the ones which are simple to describe. Here it would be simpler if we didn't have to distinguish the case where the column does not exist and the case where it does (except to describe how and where new columns are created).

What do you think?